components/script/dom/dommatrixreadonly.rs

```diff
@@ -691,14 +690,12 @@ impl DOMMatrixReadOnlyMethods for DOMMatrixReadOnly {
     }
 
     // https://drafts.fxtf.org/geometry-1/#dom-dommatrixreadonly-tofloat64array
-    #[allow(unsafe_code)]
-    fn ToFloat64Array(&self, cx: JSContext) -> NonNull<JSObject> {
-        let arr = self.matrix.borrow().to_array();
-        unsafe {
-            rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
-            let _ = Float64Array::create(*cx, CreateWith::Slice(&arr), array.handle_mut()).unwrap();
-            NonNull::new_unchecked(array.get())
-        }
+    fn ToFloat64Array(&self, cx: JSContext) -> Float64Array {
+        let vec: Vec<f64> = self.matrix.borrow().to_array().to_vec();
```

Comment:
Ok, and what about just `create_typed_array(cx, &self.matrix.borrow().to_array(), ...` to avoid allocating a new vec?

Reply:
Indeed not needed

Reply:
Great, it would be nice to have a look at the previously merged code to see if we missed similar opportunities...